src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo | null;
+  closeModal: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const [todoOwner, setTodoOwner] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const findUser = useCallback(() => {
+    setIsLoading(true);
+    if (todo?.userId) {
+      getUser(todo.userId)
+        .then(setTodoOwner)
+        .catch(error => {
+          alert(error);
+          setTodoOwner(null);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
```

Comment:
Fetch todos flow should looks like this 👍 
- Only one moment, `useCallback` doesn't make any sense, as you don't pass this function to other components